packages/blade/src/components/ActionList/ActionListItem.tsx

```diff
@@ -0,0 +1,240 @@
+import React from 'react';
+import styled from 'styled-components';
+import { StyledActionListItem } from './StyledActionListItem';
+import { componentIds } from './componentIds';
+import type { StyledActionListItemProps } from './getBaseActionListItemStyles';
+import Box from '~components/Box';
+import type { IconComponent } from '~components/Icons';
+import { useDropdown } from '~components/Dropdown/useDropdown';
+import { Text } from '~components/Typography';
+import { isReactNative, makeAccessible, makeSize } from '~utils';
+import { Checkbox } from '~components/Checkbox';
+
+const ActionListItemContext = React.createContext<{
+  intent?: ActionListItemProps['intent'];
+}>({});
+
+const SectionDivider = styled(Box)((props) => ({
+  // @TODO: replace this with token value if we add 1px token
+  height: makeSize(1),
+  backgroundColor: props.theme.colors.surface.border.normal.lowContrast,
+  margin: `${makeSize(props.theme.spacing[1])} ${makeSize(props.theme.spacing[3])}`,
+}));
+
+const StyledActionListSectionTitle = styled(Box)((props) => ({
+  padding: makeSize(props.theme.spacing[3]),
+}));
+
+const ActionListSection = ({
+  title,
+  children,
+  hideDivider,
+}: {
+  title: string;
+  children: React.ReactNode[] | React.ReactNode;
+  hideDivider?: boolean;
+}): JSX.Element => {
+  return (
+    <Box>
+      <StyledActionListSectionTitle>
+        <Text color="surface.text.muted.lowContrast" size="small" weight="bold">
+          {title}
+        </Text>
+      </StyledActionListSectionTitle>
+      {children}
+      {hideDivider ? null : <SectionDivider />}
+    </Box>
+  );
+};
+
+ActionListSection.componentId = componentIds.ActionListSection;
+
+const ActionListItemIcon = ({ icon }: { icon: IconComponent }): JSX.Element => {
+  const Icon = icon;
+  const { intent } = React.useContext(ActionListItemContext);
+  return (
+    <Icon
+      color={
+        intent === 'negative'
+          ? 'feedback.icon.negative.lowContrast'
+          : 'surface.text.muted.lowContrast'
+      }
+      size="medium"
+    />
+  );
+};
+
+const ActionListItemText = ({ children }: { children: string }): JSX.Element => {
+  return (
+    <Text variant="caption" color="surface.text.muted.lowContrast">
+      {children}
+    </Text>
+  );
+};
+
+const ActionListCheckboxWrapper = styled(Box)((_props) => ({
+  pointerEvents: 'none',
+}));
+
+const getActionListItemRole = (href?: string): 'link' | 'menuitem' | 'option' => {
```

Comment:
the way we have `accessibilityMap` shall we also create a11y roles so we can put platform dependent things at one place and then we don't need to put checks here and it can also act as constants?

Reply:
I feel it might not be consistent. Just because here we're using `menuitem` instead of `option`, does not mean every `option` role can be replaced with `menuitem`. It might also be a `listitem` in some scenarios. So I think we should handle this on case-by-case.